tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1130,26 +1135,37 @@ void Trace::record(const TraceNode& trace_node) {
     this->history.push_back(trace_node);
 }
 
-void Trace::create_replay() {
-    // Reconstruct the hugepage from the command cache
-    SystemMemoryManager& manager = this->command_queue.manager;
-    const uint32_t command_queue_id = this->command_queue.id;
-    const bool lazy_push = true;
-    for (auto& [device_command, data, command_type, num_data_bytes] : this->history) {
-        uint32_t issue_write_ptr = manager.get_issue_queue_write_ptr(command_queue_id);
-        device_command.update_buffer_transfer_src(0, issue_write_ptr + DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND);
-        manager.cq_write(device_command.data(), DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND, issue_write_ptr);
-        manager.issue_queue_push_back(DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND, lazy_push, command_queue_id);
-
-        uint32_t host_data_size = align(data.size() * sizeof(uint32_t), 16);
-        manager.cq_write(data.data(), host_data_size, manager.get_issue_queue_write_ptr(command_queue_id));
-        vector<uint32_t> read_back(host_data_size / sizeof(uint32_t), 0);
-        tt::Cluster::instance().read_sysmem(
-            read_back.data(), host_data_size, issue_write_ptr + DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND, 0, 0);
-        manager.issue_queue_push_back(host_data_size, lazy_push, command_queue_id);
+void Trace::validate() {
+    for (const auto& cmd : this->trace_queue().worker_queue) {
+        if (cmd.blocking.has_value()) {
+            TT_ASSERT(!cmd.blocking.value(), "Blocking commands are not supported in traces");
```

Comment:
this looks like it should be a TT_FATAL.  TT_ASSERTs will go away w/ release builds, API checks should use fatal